scout/adapter/mongo/omics_variant.py

```diff
@@ -35,18 +35,42 @@ def delete_omics_variants(self, case_id: str, file_type: str):
 
     def set_samples(self, case_obj: dict, omics_model: dict):
         """Internal member function to connect individuals.
-        OMICS variants do not have a genotype as such."""
+        OMICS variants do not have a genotype as such.
+        Select individuals that match on rna_sample_id if these are provided.
+        For a fallback, match on sample id (as when we are loading a pure RNA case eg),
+        or fall back to assigning the variants to an individual with affected status to have them display
+        on variantS queries.
+        ."""
 
         samples = []
 
+        sample_id = None
+        display_name = None
+
         for ind in case_obj.get("individuals"):
-            if omics_model["sample_id"] == ind.get("individual_id"):
-                sample = {
-                    "sample_id": ind["individual_id"],
-                    "display_name": ind["display_name"],
-                    "genotype_call": "./1",
-                }
-                samples.append(sample)
+            if omics_model["sample_id"] == ind.get("rna_sample_id"):
+                sample_id = ind["rna_sample_id"]
+                display_name = ind["display_name"]
+
+        if not sample_id:
```

Comment:
What if we omit the comparison above and in this patch we just take into account the ind matching on sample_id == individual_id?